site/content/blog/2020-08-28-introducing-the-graphql-add-on-for-zap/index.md

```diff
@@ -0,0 +1,108 @@
+
+![The GraphQL Options Panel](images/graphql-options-panel.png)
+
+These settings allow you to control the output of the add-on's Query Generator. Based on the combination of your chosen values for the above options, you will be able to send unique queries to the endpoint.
+
+For example, by increasing the maximum query depth, you may be able to exhaust the server's resources. Or if you specify arguments using variables, you could perform injection attacks by making use of ZAP's input vectors. You may also be able to bypass authorization and discover IDORs by generating queries with or without certain arguments and fields. The possibilities are endless!
+
+I did say before, that the add-on is still in an early stage. If you face any problems or have any feature requests, feel free to join in the discussion on the [ZAP User Group](https://groups.google.com/d/msg/zaproxy-users/j0Uxg6cUIY8/ghBP26sjBgAJ) or contact me directly. You can also ping us on the irc channel ([#zaproxy on Freenode](https://webchat.freenode.net/#zaproxy)) or raise a GitHub issue for bugs.
+
+## The Story Behind this Add-On
+
+Developing this add-on was my project this summer, as part of Google Summer of Code 2020. The summer will soon turn to autumn, but I'm going to continue working on the project. Yes, that's how fun it was! Over the course of 3 months, under the guidance of Simon, Ricardo and Rick, I was able to learn a lot. I was a complete beginner a few months ago - I didn't even know what an API was! If I was able to develop an add-on, then you can too!
+
+I primarily worked on the [zap-extensions](https://github.com/zaproxy/zap-extensions/) GitHub repository. Here are all the major pull requests I made:
+
+- PR [#2396](https://github.com/zaproxy/zap-extensions/pull/2396) - Bare bones add-on
+- PR [#2420](https://github.com/zaproxy/zap-extensions/pull/2420) - Import Dialog
+- PR [#2431](https://github.com/zaproxy/zap-extensions/pull/2431) - Import Functionality
+- PR [#2485](https://github.com/zaproxy/zap-extensions/pull/2485) - Generate GraphQL Queries from Schema
+- PR [#2521](https://github.com/zaproxy/zap-extensions/pull/2521) - Add Option for Separated Arguments
+- PR [#6044](https://github.com/zaproxy/zaproxy/pull/6044) - Call default spider filter after custom filters
+
+The released add-on package can be found [here](https://github.com/zaproxy/zap-extensions/releases/tag/graphql-v0.1.0).
+
+If you'd like to know more about the implementation details of this add-on, you can go to the [blog](https://akshathkothari.com/gsoc/) for this project. If you are also interested in contributing to ZAP, I urge you to read the "[Hacking ZAP](https://www.zaproxy.org/blog/2014-03-10-hacking-zap-1-why-should-you/)" series of blog posts or go through the recently released "[ZAP in Ten](https://www.alldaydevops.com/zap-in-ten)" videos. Finally, go check out [CONTRIBUTING.md](https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md).
```

Comment:
`https://www.zaproxy.org` could be removed to keep links relative, when testing locally.